odoo/addons/base/models/ir_actions.py

```diff
@@ -82,28 +82,31 @@ def _get_eval_context(self, action=None):
         }
 
     @api.model
-    @tools.ormcache('frozenset(self.env.user.groups_id.ids)', 'model_name')
-    def get_bindings(self, model_name):
+    @tools.ormcache('frozenset(self.env.user.groups_id.ids)', 'model_name', 'debug')
+    def get_bindings(self, model_name, debug=True):
         """ Retrieve the list of actions bound to the given model.
 
            :return: a dict mapping binding types to a list of dict describing
                     actions, where the latter is given by calling the method
                     ``read`` on the action record.
         """
-        # DLE P19: Need to flush before doing the SELECT, which act as a search.
-        # Test `test_bindings`
-        self.flush()
         cr = self.env.cr
-        query = """ SELECT a.id, a.type, a.binding_type
-                    FROM ir_actions a, ir_model m
-                    WHERE a.binding_model_id=m.id AND m.model=%s
-                    ORDER BY a.id """
-        cr.execute(query, [model_name])
         IrModelAccess = self.env['ir.model.access']
 
         # discard unauthorized actions, and read action definitions
         result = defaultdict(list)
         user_groups = self.env.user.groups_id
+        if not debug:
+            user_groups -= self.env.ref('base.group_no_one')
+
+        self.flush()
+        cr.execute("""
+            SELECT a.id, a.type, a.binding_type
+              FROM ir_actions a
+              JOIN ir_model m ON a.binding_model_id = m.id
+             WHERE m.model = %s
+          ORDER BY a.id
```

Comment:
the random indentation in this query is disturbing to me

Reply:
@KangOl Round 1. Ready?... Fight!